plugin-server/src/main/pluginsServer.ts

```diff
@@ -430,26 +431,50 @@ export async function startPluginsServer(
         }
 
         if (capabilities.sessionRecordingIngestion) {
-            const kafka = hub?.kafka ?? createKafkaClient(serverConfig as KafkaConfig)
             const postgres = hub?.postgres ?? createPostgresPool(serverConfig.DATABASE_URL)
             const teamManager = hub?.teamManager ?? new TeamManager(postgres, serverConfig)
-            const { stop, isHealthy: isSessionRecordingsHealthy } = await startSessionRecordingEventsConsumer({
+            const {
+                stop,
+                isHealthy: isSessionRecordingsHealthy,
+                join,
+            } = await startSessionRecordingEventsConsumer({
                 teamManager: teamManager,
-                kafka: kafka,
                 kafkaConfig: serverConfig as KafkaConfig,
-                partitionsConsumedConcurrently: serverConfig.RECORDING_PARTITIONS_CONSUMED_CONCURRENTLY,
                 consumerMaxBytes: serverConfig.KAFKA_CONSUMPTION_MAX_BYTES,
                 consumerMaxBytesPerPartition: serverConfig.KAFKA_CONSUMPTION_MAX_BYTES_PER_PARTITION,
                 consumerMaxWaitMs: serverConfig.KAFKA_CONSUMPTION_MAX_WAIT_MS,
             })
             stopSessionRecordingEventsConsumer = stop
+            joinSessionRecordingEventsConsumer = join
             healthChecks['session-recordings'] = isSessionRecordingsHealthy
         }
 
         if (capabilities.http) {
             httpServer = createHttpServer(healthChecks, analyticsEventsIngestionConsumer, piscina)
         }
 
+        // If session recordings consumer is defined, then join it. If join
+        // resolves, then the consumer has stopped and we should shut down
+        // everything else. Ideally we would also join all the other background
+        // tasks as well to ensure we stop the server if we hit any errors and
+        // don't end up with zombie instances, but I'll leave that refactoring
+        // for another time. Note that we have the liveness health checks
+        // already, so in K8s cases zombies should be reaped anyway, albeit not
```

Comment:
note: the hobby deploy won't restart the process here. I don't think it's a blocker to merge, let's create a task to follow-though though.

One easy way to close this loop would be to have an optional loop that runs the same healthcheck logic as the liveness endpoint, and exit(1), so that docker can restart us. Not huge prio though!

Reply:
Yeah I haven't thought too much about hobby here 🤔 

My thinking is hobby we have a different level of expectation and we should target k8s as the resilient way to run it.

I think I might have missed some bits re. error handling tbh. I'm sure there's something that I should be hooking into that would identify existential errors from librdkafka. #testinproduction 😬

Reply:
Yep, not blocking to merge, with better knowledge of the failure modes we'll iterate on this!